### PR TITLE
update add model UI to include required inputs

### PR DESCRIPTION
**Pull request by ajshedivy**
## Description

Proposed fix for #2323 

Update UI to add required input fields for providers. To do this, I am leveraging `collectInputFor` field. Note that there may be a better way to do this, but opening this PR for reference 😄 

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

image
image

## Testing

[ For new or modified features, provide step-by-step testing instructions to validate the intended behavior of the change. ]

### CODE REVIEW

File: gui/src/forms/AddModelForm.tsx [hunk 1]
```diff
@@ -49,8 +50,15 @@ function AddModelForm({
...
 
-    const apiKey = formMethods.watch("apiKey");
-    return typeof apiKey === "undefined" || apiKey.length === 0;
+    const required = selectedProvider.collectInputFor
+      .filter((input) => input.required)
+      .map((input) => {
+        const value = formMethods.watch(input.key);
+        return value;
+      });
+    console.log("Required fields values:", required);
```
Comment (Patrick-Erichsen) [inline gui/src/forms/AddModelForm.tsx]:
```suggestion
    console.debug("Required fields values:", required);
```

We should document this somewhere but using `console.debug` logs to reduce noise

**Comment by Patrick-Erichsen** [comment]
This looks perfect to me! Thanks for opening this so quickly. Good to merge unless there's anything else you wanted to edit.

**Comment by ajshedivy** [comment]
@Patrick-Erichsen opening for review!

File: gui/src/forms/AddModelForm.tsx (hunk 1 above)
Comment (ajshedivy) [reply gui/src/forms/AddModelForm.tsx]: done

**Pull request merged**